conformance/utils/roundtripper/roundtripper.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package roundtripper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"time"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// RoundTripper is an interface used to make requests within conformance tests.
+// This can be overridden with custom implementations whenever necessary.
+type RoundTripper interface {
+	CaptureRoundTrip(Request) (*CapturedRequest, *CapturedResponse, error)
+}
+
+// Request is the primary input for making a request.
+type Request struct {
+	URL      url.URL
+	Host     string
+	Protocol *v1alpha2.ProtocolType
+	Method   *v1alpha2.HTTPMethod
```

Comment:
Use string for Protocol/Method - simpler API. This doesn't seem to have an any advantage.

Reply:
Updated, thanks!